Mathlib/Analysis/Asymptotics/TVS.lean

```diff
@@ -81,17 +81,65 @@ def IsLittleOTVS (𝕜 : Type*) {α E F : Type*}
 @[inherit_doc]
 notation:100 f " =o[" 𝕜 ";" l "] " g:100 => IsLittleOTVS 𝕜 l f g
 
-variable {α β 𝕜 E F : Type*}
+variable {α β 𝕜 E F G : Type*}
 
 section TopologicalSpace
 
 variable [NontriviallyNormedField 𝕜]
   [AddCommGroup E] [TopologicalSpace E] [Module 𝕜 E]
   [AddCommGroup F] [TopologicalSpace F] [Module 𝕜 F]
+  [AddCommGroup G] [TopologicalSpace G] [Module 𝕜 G]
+
+section congr
+
+variable {f f₁ f₂ : α → E} {g g₁ g₂ : α → F} {l : Filter α}
+
+theorem isLittleOTVS_congr (hf : f₁ =ᶠ[l] f₂) (hg : g₁ =ᶠ[l] g₂) :
+    f₁ =o[𝕜;l] g₁ ↔ f₂ =o[𝕜;l] g₂ := by
+  simp only [IsLittleOTVS]
+  refine forall₂_congr fun U hU => exists_congr fun V => and_congr_right fun hV =>
+    forall₂_congr fun ε hε => Filter.eventually_congr ?_
+  filter_upwards [hf, hg] with _ e₁ e₂
+  rw [e₁, e₂]
+
+/-- A stronger version of `IsLittleOTVS.congr` that requires the functions only agree along the
+filter. -/
+theorem IsLittleOTVS.congr' (h : f₁ =o[𝕜;l] g₁) (hf : f₁ =ᶠ[l] f₂) (hg : g₁ =ᶠ[l] g₂) :
+    f₂ =o[𝕜;l] g₂ :=
+  (isLittleOTVS_congr hf hg).mp h
+
+theorem IsLittleOTVS.congr (h : f₁ =o[𝕜;l] g₁) (hf : ∀ x, f₁ x = f₂ x) (hg : ∀ x, g₁ x = g₂ x) :
+    f₂ =o[𝕜;l] g₂ :=
+  h.congr' (univ_mem' hf) (univ_mem' hg)
+
+theorem IsLittleOTVS.congr_left (h : f₁ =o[𝕜;l] g) (hf : ∀ x, f₁ x = f₂ x) : f₂ =o[𝕜;l] g :=
+  h.congr hf fun _ => rfl
+
+theorem IsLittleOTVS.congr_right (h : f =o[𝕜;l] g₁) (hg : ∀ x, g₁ x = g₂ x) : f =o[𝕜;l] g₂ :=
+  h.congr (fun _ => rfl) hg
```

Comment:
I'm not sure that we need these non-filter `congr` lemmas. Should we use `EventuallyEq` everywhere?